pandas/tslib.pyx

```diff
@@ -291,10 +291,25 @@ class Timestamp(_Timestamp):
     def combine(cls, date, time):
         return cls(datetime.combine(date, time))
 
-    def __new__(cls, object ts_input, object offset=None, tz=None, unit=None):
+    def __new__(cls,
+            # Keep legacy parameter names.
+            object ts_input, object offset=None, tz=None, unit=None,
+            # Switch to datetime positional paramers, starting with `minute`.
+            minute=0, second=0, microsecond=0, tzinfo=None):
```

Comment:
`datetime.datetime` raises `TypeError` when `None` is passed for any argument (except `tzinfo`), which led me to default to `0`. Is that ok?

```
----> 1 dt.datetime(2012, 1, 1, None)

TypeError: an integer is required (got type NoneType)
```